Resources/public/js/views/tabs/ez-locationviewlocationstabview.js

```diff
@@ -141,6 +144,62 @@ YUI.add('ez-locationviewlocationstabview', function (Y) {
          */
         _enableSetMainLocationRadios: function () {
             this.get('container').all('.ez-main-location-radio').set('disabled', false);
+        },
+
+        /**
+         * Tap event handler on the `Remove selected` button. It fires the
+         * `removeLocations` event
+         *
+         * @method _removeSelectedLocations
+         * @protected
+         * @param {EventFacade} e
+         */
+        _removeSelectedLocations: function (e) {
+            var c = this.get('container'),
+                locations = [];
+
+            c.all('.ez-location-checkbox').each(function (checkbox) {
+                if (checkbox.get('checked')) {
+                    locations.push(checkbox.getAttribute('data-location-id'));
+                }
+            });
+
+            if (locations.length > 0) {
+                this._disableLocationsInputs();
+                this.fire('removeLocations', {
+                    locations: locations,
+                    afterRemoveLocationsCallback: Y.bind(this._refresh, this),
+                    cancelRemoveLocationsCallback: Y.bind(this._enableLocationsInputs, this)
```

Comment:
I would pass only one callback and make sure it receives the results of the removal so that it can either refresh the tab or enable back the checkboxes.